core/amber/src/main/python/core/models/tuple.py

```diff
@@ -245,9 +245,10 @@ def finalize(self, schema: Schema) -> None:
         :param schema: target Schema to finalize the Tuple.
         :return:
         """
-        self.cast_to_schema(schema)
```

Comment:
also, in the cast_to_schema, if the field is a BINARY type and the value is already bytes type, we are skipping the serialization. This seems to be contradictory to our understanding of the cause of the bug?

Reply:
I don't think so. We only need to perform serialization and deserialization for users if they haven't done any serialization yet.

Reply:
ok